usr/src/uts/common/fs/zfs/range_tree.c

```diff
@@ -324,16 +325,29 @@ range_tree_find(range_tree_t *rt, uint64_t start, uint64_t size)
 	return (NULL);
 }
 
+/*
+ * Given an extent start offset and size, will look through the provided
+ * range tree and find a suitable start offset (starting at `start') such
+ * that the requested extent _doesn't_ overlap with any range segment in
+ * the range tree.
+ */
+uint64_t
+range_tree_find_gap(range_tree_t *rt, uint64_t start, uint64_t size)
+{
+	range_seg_t *rs;
+	while ((rs = range_tree_find_impl(rt, start, size)) != NULL)
+		start = rs->rs_end;
+	return (start);
+}
+
 void
 range_tree_verify(range_tree_t *rt, uint64_t off, uint64_t size)
 {
 	range_seg_t *rs;
 
-	mutex_enter(rt->rt_lock);
 	rs = range_tree_find(rt, off, size);
```

Comment:
This is a change suggested by Tim Chase. The purpose is to make range_tree_verify the same as all other range_tree functions, i.e. to make locking external to the function. The only usage of this function is in metaslab_check_free, where all calls to range_tree_verify got bracketed by one general acquisition of the lock.